meter/powerwall.go

```diff
@@ -149,3 +211,20 @@ func (m *PowerWall) batterySoc() (float64, error) {
 
 	return float64(res.Percentage), err
 }
+
+// decorate soc
+func (m *PowerWall) socG() (float64, error) {
+	ess, err := m.energySite.EnergySiteStatus()
+	if err != nil {
+		return 0, err
+	}
+	currentSoc := math.Round(ess.PercentageCharged + 0.5) // .5 ensures we round up
+	return currentSoc, nil
+}
+
+func (m *PowerWall) limitSocS(limit float64) error {
+	if err := m.energySite.SetBatteryReserve(uint64(limit)); err != nil {
```

Comment:
```suggestion
	return m.energySite.SetBatteryReserve(uint64(limit))
```

Das kannst Du auch im New in eine Closure packen.